src/theory/strings/theory_strings.cpp

```diff
@@ -623,15 +614,17 @@ void TheoryStrings::notifyFact(TNode atom,
     Node pc = d_state.getPendingConflict();
     if (!pc.isNull())
     {
-      std::vector<Node> a;
-      a.push_back(pc);
       Trace("strings-pending")
           << "Process pending conflict " << pc << std::endl;
-      Node conflictNode = d_im.mkExplain(a);
+      InferInfo iiPrefixConf;
+      iiPrefixConf.d_id = Inference::PREFIX_CONFLICT;
+      iiPrefixConf.d_conc = d_false;
+      utils::flattenOp(AND, pc, iiPrefixConf.d_ant);
       Trace("strings-conflict")
-          << "CONFLICT: Eager prefix : " << conflictNode << std::endl;
+          << "CONFLICT: Eager prefix : " << pc << std::endl;
       ++(d_statistics.d_conflictsEagerPrefix);
-      d_im.conflict(conflictNode);
+      // call the inference manager to send the conflict
+      d_im.processConflict(iiPrefixConf);
```

Comment:
I think that it would be better to make the pending conflict store an `InferInfo`. It is a bit confusing that a pending conflict can only be a prefix conflict and may change in the future.

Reply:
Done.